modules/Bio/EnsEMBL/PipeConfig/GenericMart_conf.pm

```diff
@@ -51,13 +51,19 @@ sub default_options {
         'species'     => [],
         'antispecies' => [],
         'division'    => '',
-        'mart_dir'    => getcwd,
-        'scratch_dir' => catdir('/hps/scratch/flicek/ensembl', $self->o('env_user'), $self->o('pipeline_name')),
-        'test_dir'    => catdir('/hps/nobackup/flicek/ensembl/production', $self->o('env_user'),
-            'mart_test', $self->o('pipeline_name')),
+        'mart_dir'    => getcwd
     }
 }
 
+sub pipeline_wide_parameters {
+    my ($self) = @_;
+    return {
+        %{$self->SUPER::pipeline_wide_parameters},
+        'scratch_dir' => catdir('/hps/scratch/flicek/ensembl', $self->o('env_user'), $self->o('pipeline_name')),
```

Comment:
I don't think this is the best way to set these parameters - the standard approach is to set the default values in the 'default_options' method (ie as was done in the now-deleted lines 55-56), and then set those values in the pipeline_wide_parameters. E.g. https://github.com/Ensembl/ensembl-production/blob/release/105/modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/ProteinFeatures_conf.pm#L314

The standard approach means that, if we want to, we can override the paths for a given instance of the pipeline - that's not possible here.

Reply:
Yes indeed, but the PopulateMart.pm is not been given the parameters, then using its own default set to simply `/scratch/`
I wanted to avoid passing parameter to PopulateMart module.

Reply:
I don't disagree with setting the pipeline-wide parameters, just with how it's implemented.
The `scratch_dir` and `test_dir` parameters should remain in the `default_options` method, and here we would have
```
scratch_dir => $self->o('scratch_dir'),
test_dir => $self->o('test_dir'),
```

Reply:
Ah! ok ! got it!

Reply:
Hope that's ok now.